### PR TITLE
reorg: move js runtime tests to cli/js/tests/

### PR DESCRIPTION
**Pull request by bartlomieju**
All Deno runtime test code was moved to cli/js/tests/ directory.
It makes a clear distinction that cli/js/tests/ contains code
that is run under Deno runtime as opposed to code in cli/js/ which
is used to create bundle and snapshot with "deno_typescript".

### CODE REVIEW

**Comment by bartlomieju** [comment]
Ref #4248 
Addresses point d)
> >d) Move runtime test code to a separate directory. I'm not really sure where exactly this code should live, but moving it to separate location appears to be a good idea to point out that testing files are valid Deno programs. Ultimately we should strive to make test each file self-contained to allow making each op a standalone thing (it won't be possible at the moment).

>Agree - with its own readme explaining that the tests run under the Deno runtime as opposed to being bundled & snapshotted via deno_typescript. cli/js/tests is my first instinct.

**Comment by bartlomieju** [comment]
#4302 should land first; then merge `master` branch into this PR to ensure everything still passes

File: cli/js/tests/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  ensuring that all started async ops are done before test finishes
+
+`unit_test_runner.ts` is main script used to run unit tests.
+
+Runner discoveres required permissions combinations by loading
+`cli/js/tests/unit_tests.ts` and going through all registered instances of
+`unitTest`. For each discovered permission combination a new Deno process is
+created with respective `--allow-*` flags which loads
+`cli/js/tests/unit_tests.ts` and executes all `unitTest` that match runtime
+permissions.
```
Comment (ry) [inline cli/js/tests/README.md]: \o/

**Comment by bartlomieju** [comment]
> I'm a little concerned about removing the _test.ts suffix from the files. These wouldn't match under `deno test`'s current logic.

I had similar concerns, so I guess I'm just gonna revert suffix removal.

> Unintentional deno_typescript/typescript changes

:+1: thanks

File: cli/js/tests/dom_iterable_test.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
-import { unitTest, assert, assertEquals } from "../test_util.ts";
+import { unitTest, assert, assertEquals } from "./test_util.ts";
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 function setup() {
```
Comment (ry) [inline cli/js/tests/dom_iterable_test.ts]: so glad to be rid of this dir

**Pull request merged**